core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/handlers/HandlerService.java

```diff
@@ -94,15 +94,15 @@ private static void check(Set<Class> handlerTargets, Class target) {
     return (Handler<T>) handler;
   }
 
-  public <T extends ManagedType> Specification handleInsert(T managedType) {
+  public <T extends Modeled> Specification handleInsert(T managedType) {
     try {
       return getHandler(managedType).handleInsert(managedType);
     } catch (Exception e) {
       throw new ValidationException(e);
     }
   }
 
-  public Specification handleUpdate(ManagedType managedType, ManagedType existing) {
+  public Specification handleUpdate(Modeled managedType, Modeled existing) {
```

Comment:
Parameter names don't match the type. I believe `Entity` or similar variant would convey more information about what it is here. `Modeled` (and `ManagedType`) are too abstract.

Reply:
Happy to hear suggestions of what would convey the intent better than Modeled.

To me a 'Modeled' Object is one that exists in the core Model. It has a state and a specification. What should we call it instead?

Reply:
will rename the parameters. Good spot.